ocaml/forkexecd/helper/vfork_helper.c

```diff
@@ -335,14 +335,43 @@ reset_signal_handlers(void)
 static void
 clear_cgroup(void)
 {
-    int fd = open("/sys/fs/cgroup/systemd/cgroup.procs", O_WRONLY|O_CLOEXEC);
-    if (fd >= 0) {
-        char string_pid[32];
-        int ignored __attribute__((unused));
-        sprintf(string_pid, "%d\n", (int) getpid());
-        ignored = write(fd, string_pid, strlen(string_pid));
+    // list of files to try, terminated by an empty string
+    static const char cgroup_files[] =
+        "/sys/fs/cgroup/systemd/cgroup.procs\0"
+        "/sys/fs/cgroup/cgroup.procs\0"
+        "\0";
+
+    char string_pid[32];
+    int last_error = 0;
+    const char *last_error_msg = NULL;
+    const char *last_fn = NULL;
+
+    sprintf(string_pid, "%d\n", (int) getpid());
```

Comment:
I'd like to see `snprintf`

Reply:
Done, not that it matters much that string format is not able to overflow that buffer.